packages/adapter/addon/-private/build-url-mixin.ts

```diff
@@ -55,7 +58,13 @@ export default Mixin.create({
     @param {Object} query object of query parameters to send for query requests.
     @return {String} url
   */
-  buildURL(modelName, id, snapshot, requestType, query) {
+  buildURL(
+    modelName: string,
+    id: string | string[] | Record<string, any> | null,
```

Comment:
The `Record` would be very confusing for the data repo, so we use Dict and ConfidentDict instead